### PR TITLE
Upgrade mesos in itests to 0.24.1

### PR DESCRIPTION
**Pull request by nhandler**
This is the first step in upgrading to mesos 0.24.1.

### CODE REVIEW

**Comment by solarkennedy** [comment]
Nice. Ship if pass.

**Comment by Rob-Johnson** [comment]
@nhandler thanks for driving this, but I'm less sure of this one:

marathon wasn't tested against mesos 0.24.* until it hit 0.13.0 - we're still on 0.11.1 and blocked by marathon-python, so I'm wary of going any further.

I guess we could run the marathon test suite against mesos 0.24.1 and see what happens?

**Comment by Rob-Johnson** [comment]
ok, first run of marathon tests failed:

```
[2016-02-05 12:00:01,398] INFO marathon: [DEBUG] [02/05/2016 11:59:59.814] [marathon-akka.actor.default-dispatcher-12] [akka://marathon/user/rateLimiter/1] received handled message CleanupOverdueDelays (mesosphere.marathon.integration.setup.ProcessKeeper$:108)
[2016-02-05 12:00:02,463] INFO mesos: I0205 12:00:02.019701  3215 slave.cpp:4179] Querying resource estimator for oversubscribable resources (mesosphere.marathon.integration.setup.ProcessKeeper$:108)
[info] - create a simple app without health checks via secondary (proxying) *** FAILED ***
[info]   java.lang.IllegalStateException: Process marathon does not came up within time bounds (30 seconds). Give up. 
/huge stacktrace
```

I'm trying again. This takes... so long.

**Comment by solarkennedy** [comment]
The failures now look like python-mesos can't interpret the new state.json endpoint stuff. I'm not sure what we are going to do with that.

**Comment by nhandler** [comment]
The issue had to do with mesos.cli being broken in mesos 0.24.x (https://issues.apache.org/jira/browse/MESOS-3556). I have gone ahead and upgraded us to mesos.cli 0.1.5, which was created to resolve this issue. This is likely the last mesos.cli release (it is deprecated in favor of dcoscli. If a future mesos upgrade breaks mesos.cli, we will either need to fork it or migrate to dcoscli. But for now, we are fine.

Tests are all green.

File: yelp_package/dockerfiles/itest/mesos/Dockerfile [hunk 1]
```diff
@@ -15,4 +15,4 @@
 FROM ubuntu:trusty
 RUN echo "deb http://repos.mesosphere.com/ubuntu trusty main" > /etc/apt/sources.list.d/mesosphere.list
 RUN apt-key adv --keyserver keyserver.ubuntu.com --recv 81026D0004C44CF7EF55ADF8DF7D54CBE56151BF
-RUN apt-get update && apt-get -y install mesos=0.23.1-0.2.61.ubuntu1404
+RUN apt-get update && apt-get -y install mesos=0.24.1-0.2.35.ubuntu1404
```
Comment (solarkennedy) [inline yelp_package/dockerfiles/itest/mesos/Dockerfile]: Can you actually undo *these* changes (the upgraded mesos parts) first? 

To answer the question of: will this bump of mesos-cli work on our existing infra?

Comment (solarkennedy) [inline yelp_package/dockerfiles/itest/mesos/Dockerfile]: If yes, *then* ship + bump + ship er whatever.

**Comment by solarkennedy** [comment]
Ship!

**Comment by nhandler** [comment]
Ok. I've reverted the mesos upgrade portion of this change. Now, it is basically a mesos.cli upgrade + some transition prep work to make the eventual mesos upgrade work fine. I'll roll a new release for this change, let it sit, and then prepare a new PR for the mesos upgrade.

**Pull request merged**